### PR TITLE
Adding properties grid enhancements in execution plan

### PR DESCRIPTION
**Pull request by aasimkhan30**
image
Adding a popup viewer for long texts so that people can select and copy from them. 
image

Also fixed an issue with the execution plan editor caching

### CODE REVIEW

**Comment by alanrenmsft** [comment]
do you really need to add a custom text viewer? a tooltip could work even better in my opinion since you can see all text at once with text wrapping.

File: src/sql/workbench/contrib/executionPlan/browser/LongTextPopupViewer.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+		this._text = DOM.$('span');
+		this._wrapper.tabIndex = 0;
+		this._wrapper.appendChild(this._text);
+		document.body.appendChild(this._wrapper);
+		this.position((<any>this._args).position);
+
+
+		this._wrapper.addEventListener('keydown', (e: KeyboardEvent) => {
+			let eventHandled = false;
+			if (e.key === 'Enter' || e.key === 'Escape') {
```
Comment (lewis-sanchez) [inline src/sql/workbench/contrib/executionPlan/browser/LongTextPopupViewer.ts:36]: Why does "enter" cancel changes?

**Comment by lewis-sanchez** [comment]
Looks good to me, but please address Alan's comment before merging.

File: src/sql/workbench/contrib/executionPlan/browser/LongTextPopupViewer.ts (hunk 1 above)
Comment (aasimkhan30) [reply src/sql/workbench/contrib/executionPlan/browser/LongTextPopupViewer.ts:36]: Because there are no changes in the first place. It is just a viewer

**Review by lewis-sanchez — APPROVED** [review]

**Comment by aasimkhan30** [comment]
> do you really need to add a custom text viewer? a tooltip could work even better in my opinion since you can see all text at once with text wrapping.

@alanrenmsft , the tooltips do not let the users select a portion of text to copy. This gives the users more flexibility on what they want to copy. 

@erinstellato-ms, what do you think about this change?

**Comment by alanrenmsft** [comment]
@aasimkhan30, To me it doesn't see to be a big issue, user can simply copy the whole thing to an text editor which is much more convinient to find a part from a long string.
 more importantly, is this feature accessible?

**Comment by erinstellato-ms** [comment]
@aasimkhan30 @alanrenmsft I don't know the specifics of how this is implemented, but the idea is to provide the capability for users to select and copy the text.  Ideally this is a one-step operation.

**Comment by erinstellato-ms** [comment]
@alanrenmsft I sync'd with @aasimkhan30 offline and we reviewed the changes.  There are multiple options for users now in terms of copy:

1. click on the cell, use CTRL + C
2. click on the cell, right-click and select copy
3. click on the cell, press Enter, select some or all text and copy

Bonus: multiple cells can be selected and copied.  I'm good with this change.

**Comment by alanrenmsft** [comment]
@erinstellato-ms how about the keyboard only experience?

@aasimkhan30 Can we use existing slickgrid editor experience to achieve the same? you can make the editor read-only.

**Comment by erinstellato-ms** [comment]
@aasimkhan30 To confirm, by "keyboard only experience", you mean a user navigating within the Properties window (without a mouse)?  We didn't test that.  If you can navigate with tab in the Properties window, and then with a cell selected use CTRL + C (or enter and then CTRL + A and CTRL +C), I would deem that acceptable.  However, I don't know if SHIFT + arrow key would then allow you to select multiple cells, and then do CTRL + C.  Let me know if there are additional scenarios.

**Pull request merged**